yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -285,7 +285,7 @@ export class AvmPersistableStateManager {
   public writeL1Message(recipient: EthAddress | Fr, content: Fr) {
     this.log.debug(`L1Messages(${recipient}) += ${content}.`);
     const recipientAddress = recipient instanceof EthAddress ? recipient : EthAddress.fromField(recipient);
-    const message = new L2ToL1Message(recipientAddress, content);
+    const message = new L2ToL1Message(recipientAddress, content, 0);
```

Comment:
Is this functional? I.e., will it automatically work once we switch the publiccontext to the avm?

Reply:
It will. We don't really need counters for messages created in public functions. It's added to the L2ToL1Message struct because the private tail needs to use counters to split them into revertible and non-revertible sets. Eventually, the avm can silo the value and only push the final hash to the public inputs. Then we don't need to share the same struct for the messages in private and public accumulated data.